extensions/fail.md

```diff
@@ -0,0 +1,50 @@
+---
+title: "`FAIL` Message"
+layout: spec
+copyrights:
+  -
+    name: "Daniel Oaks"
+    period: "2019"
+    email: "daniel@danieloaks.net"
+---
+
+This document specifies the `FAIL` message, intended to provide a clean, consistent interface for sending general errors to clients. We want implementors to be able to provide appropriate error information for newly-developed commands and functions without having to reserve new numerics.
+
+The `FAIL` message may indicate a complete failure to process a given command or function, or simply something about how the processing goes that clients should be aware of.
+
+
+## Message Format
+
+This message has the following format:
+
+    FAIL <command> [<argument>...] <error-code> <description>
```

Comment:
How do you suggest parsing error-code which comes after free form arguments?
Is this `FOO` another argument, or it's a error code already?

Reply:
Second-last parameter is the code and the last param is description, anything between those and the first param are just arguments. Once peeps parse the line into a message format hopefully shouldn't be a big drama to strip off the last two params and process the args, or however they want to do it.

Reply:
It would be a lot cleaner and easier to parse if error-code came directly after the command. The arguments may not always be important but the command and error-code are always important.

Reply:
Hmm... yeah that would definitely make a lot of sense.

Reply:
This is now done